tokio/src/util/idle_notified_set.rs

```diff
@@ -0,0 +1,463 @@
+//! This module defines an `IdleNotifiedSet`, which is a collection of elements.
+//! Each element is intended to correspond to a task, and the collection will
+//! keep track of which tasks have notified their waker, and which have not.
```

Comment:
nit, take it or leave it: should this maybe be rephrased as 
```suggestion
//! keep track of which tasks have had their wakers notified, and which have not.
```
(it feels like the current phrasing kind of implies that it tracks whether the tasks have woken their _own_ waker, while in actuality, the wakeups may also be external to the task)